File: src/dataset/cpp/armstrong-numbers_exercism-Richard-m-j/gemini/GeminiFlash_armstrong_numbers_v1.cpp
#include "armstrong_numbers.h"
#include <cmath>
#include <numeric>
#include <vector>

namespace armstrong_numbers {

bool is_armstrong_number(int number) {
  if (number < 0) {
    return false;
  }

  if (number == 0) {
    return true;
  }

  std::vector<int> digits;
  int temp = number;
  while (temp > 0) {
    digits.push_back(temp % 10);
    temp /= 10;
  }

  int num_digits = digits.size();

  long long sum = 0;
  for (int digit : digits) {
    long long power = 1;
    for (int i = 0; i < num_digits; ++i) {
      power *= digit;
    }
    sum += power;

    if (sum > number) return false;
  }

  return sum == number;
}

} // namespace armstrong_numbers

File: src/dataset/Java/armstrong-numbers_exercism-awsaf174/gemini/GeminiFlash_ArmstrongNumbers_v1.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        if (numberToCheck < 0) {
            return false;
        }

        int num = numberToCheck;
        int sum = 0;
        int digits = 0;
        int temp = numberToCheck;

        while (temp > 0) {
            digits++;
            temp /= 10;
        }

        temp = numberToCheck;
        while (temp > 0) {
            int digit = temp % 10;
            sum += Math.pow(digit, digits);
            temp /= 10;
        }

        return sum == numberToCheck;
    }
}